python/MooseDocs/extensions/navigation.py

```diff
@@ -120,6 +126,38 @@ def postRender(self, page, result):
         self._addTitle(head, root, page)
         self._addName(nav, page)
 
+    def postTokenize(self, page, ast):
+
+        index = []
+        title = None
+        for head in moosetree.findall(ast, lambda n: n.name == 'Heading'):
+            if (head['level'] == 1) and (title is None):
+                title = head.text()
+
+            id_ = head.get('id')
+            if id_ == '':
+                id_ = head.text('-').lower()
+            index.append(dict(title=title, text=head.text(), bookmark=id_))
+        page['search'].extend(index)
+
+
+    def postExecute(self):
+        """Build the JSON file containing the index data."""
```

Comment:
Would it be worthwhile to expand this comment to include some information about the use cases for the apps that don't use the google search functionality?